content/blog/page-view-counter/index.md

```diff
@@ -0,0 +1,273 @@
+
+Here are the first 3 steps in code:
+
+```csharp
+    const string tableName = "viewcountertable";
+    
+    var storageAccount = CloudStorageAccount.Parse($"{yourStorageAccountConnectionString}");
+    var tableClient = storageAccount.CreateCloudTableClient();
+    CloudTable table = tableClient.GetTableReference(tableName);
+    
+    await table.CreateIfNotExistsAsync(); // we can let our code create the table if needed
+```
+
+We can now do transactions to the table! Let's get the URL parameter from the body first:
+
+```csharp
+    string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+    dynamic data = JsonConvert.DeserializeObject(requestBody);
+    string pageViewURL = data?.URL;
+```
+
+And now, we can use `URL` to create a `ViewCount` object that uses it as a key. The logic should be easy to follow but here's a high-level overview:
+
+1. Try to retrieve the `ViewCount` object for the URL.
+1. If it doesn't exist, create a new `ViewCount` object (with count 0 by default) for that URL.
+1. Increase the view count on that object by 1.
+1. Store the updated entity.
+1. Return the view count to the function caller.
+
+Here's what the code looks like:
+
+```csharp
+     var retrievedResult = table.Execute(TableOperation.Retrieve<ViewCount>(pageViewURL, "visits")); // try to retrieve count for the URL
+     var pageViewCount = (ViewCount) retrievedResult.Result; // wait for the operation and get the result into a ViewCount object 
+  
+     pageViewCount = pageViewCount ?? new ViewCount(pageViewURL); // if the entity didn't exist, create a new one with count 0
+     pageViewCount.Count++; // augment the view count by one
+     
+     table.Execute(TableOperation.InsertOrReplace(pageViewCount)); // insert or replace the entity with the new count
+     return (ActionResult)new OkObjectResult(pageViewCount.Count.ToString()); // return the view count 
+```
+
+Let's also make sure we return an error code if we don't give a "URL" identifier to the function before trying to retrieve an entity:
+
+```csharp
+  if (pageViewURL == null)
+  {
+    return (ActionResult)new StatusCodeResult(503);
+  }
+```
+The function is now ready! Let's make sure it works by running it locally and performing POST requests that have a "URL" parameter in the body.
+
+### Deploying and implementation
+
+We are now ready to deploy the Azure Function. Something that you might want to make sure of, is that you do not usually want to keep any connection string in code, so we will be getting it through the [App Settings](https://docs.microsoft.com/en-us/azure/azure-functions/functions-how-to-use-azure-function-app-settings#settings):
+
+```csharp
+  // we will have a StorageConnectionString field in the app settings where we store the storage account connection string
+  var storageAccountConnectionString = Environment.GetEnvironmentVariable("StorageConnectionString");
+```
+The App Settings can be changed in the Azure Portal at runtime.  
```

Comment:
How do you interact with these Env variables in Visual Studio code? (i figure they are regular Env variables but since these things are probably set up in Visual Studio code instead of being set in console we should talk about how they are setup and made accessible to the app locally)